lib/urlwatch/handler.py

```diff
@@ -216,3 +238,59 @@ def parse_urls_txt(urls_txt):
 
     return jobs
 
+def parse_urls_yaml(urls_yaml):
+    jobs = []
+
+    # Security checks for shell jobs - only execute if the current UID
+    # is the same as the file/directory owner and only owner can write
+    allow_shelljobs = True
+    shelljob_errors = shelljob_security_checks(urls_yaml)
+    if shelljob_errors:
+        allow_shelljobs = False
+
+    with open(urls_yaml, 'r') as f:
+        yaml_jobs = yaml.load_all(f)
+        for job in yaml_jobs:
+            # entries with all parameters commented out
+            if job is None:
+                continue
+            # set all supported keys here, so we don't need to care later
+            for key in ['name', 'data']:
+                if not key in job:
+                    job[key] = None
+
+            if "url" in job:
+                jobs.append(UrlJob(job['url'], name=job['name'], data=job['data']))
+            elif "command" in job:
+                if allow_shelljobs:
+                    jobs.append(ShellJob(job['command'], name=job['name']))
+                else:
+                    print >>sys.stderr, '\n  SECURITY WARNING - Cannot run shell jobs:\n'
+                    for error in shelljob_errors:
+                        print >>sys.stderr, '    ', error
+                    print >>sys.stderr, '\n  Please remove shell jobs or fix these problems.\n'
+                    sys.exit(1)
+            else:
+                log.warning("Entry has neither url nor command! Ignoring: %r", job)
+
+    return jobs
+
+def create_urls_yaml(jobs, urls_yaml):
+    yaml_jobs = []
+    with os.fdopen(os.open(urls_yaml, os.O_WRONLY | os.O_CREAT, 0o0644), 'w') as f:
```

Comment:
No problem. It's your coude.
But this happens on my Ubuntu with default umask settings:
```
enno@dharma ~/git/urlwatch[extendable_urls_file*]$ cat ~/.urlwatch/urls.txt

# This is an example urls.txt file for urlwatch
# Empty lines and lines starting with "#" are ignored

http://www.dubclub-vienna.com/
http://www.openpandora.org/developers.php
#http://www.statistik.tuwien.ac.at/lv-guide/u107.369/info.html
#http://www.statistik.tuwien.ac.at/lv-guide/u107.369/blatter.html
#http://www.dbai.tuwien.ac.at/education/dbs/current/index.html
#http://www.dbai.tuwien.ac.at/education/dbs/current/uebung.html
http://ti.tuwien.ac.at/rts/teaching/courses/systems_programming
http://ti.tuwien.ac.at/rts/teaching/courses/systems_programming/labor
http://ti.tuwien.ac.at/rts/teaching/courses/betriebssysteme
#http://www.complang.tuwien.ac.at/anton/lvas/effiziente-programme.html
#http://www.complang.tuwien.ac.at/anton/lvas/effizienz-aufgabe08/
http://www.kukuk.at/ical/events
http://guckes.net/cal/

# You can use the pipe character to "watch" the output of shell commands
|ls -al ~

# If you want to use spaces in URLs, you have to URL-encode them (e.g. %20)
http://example.org/With%20Spaces/

# You can do POST requests by writing the POST data behind the URL,
# separated by a single space character. POST data is URL-encoded.
http://example.com/search.cgi button=Search&q=something&category=4

enno@dharma ~/git/urlwatch[extendable_urls_file*]$ ll ~/.urlwatch/
insgesamt 24
drwxrwxr-x 2 enno enno 4096 Dez 12 01:16 cache
drwxrwxr-x 2 enno enno 4096 Okt  4 13:16 lib
-rw-r--r-- 1 enno enno 1218 Dez 12 01:04 urls.txt
-rw-rw-r-- 1 enno enno 1218 Okt  4 13:16 urls.txt.example
-rw-rw-r-- 1 enno enno 1514 Okt  4 13:22 urls.yaml.example
enno@dharma ~/git/urlwatch[extendable_urls_file*]$ gvim lib/urlwatch/handler.py
enno@dharma ~/git/urlwatch[extendable_urls_file*]$ ./urlwatch 

Found old urls.txt and imported it into /home/enno/.urlwatch/urls.yaml

An example for urls.yaml is available in /home/enno/.urlwatch/urls.yaml.example
Old urls.txt has been renamed to urls.txt.imported


You can also create /home/enno/.urlwatch/lib/hooks.py
An example is available in /home/enno/.urlwatch/lib/hooks.py.example

enno@dharma ~/git/urlwatch[extendable_urls_file*]$ ll ~/.urlwatch/
insgesamt 28
drwxrwxr-x 2 enno enno 4096 Dez 12 01:16 cache
drwxrwxr-x 2 enno enno 4096 Okt  4 13:16 lib
-rw-rw-r-- 1 enno enno 1218 Okt  4 13:16 urls.txt.example
-rw-r--r-- 1 enno enno 1218 Dez 12 01:04 urls.txt.imported
-rw-rw-r-- 1 enno enno  528 Jan  3 19:26 urls.yaml
-rw-rw-r-- 1 enno enno 1514 Okt  4 13:22 urls.yaml.example
enno@dharma ~/git/urlwatch[extendable_urls_file*]$ ./urlwatch 

  SECURITY WARNING - Cannot run shell jobs:

     /home/enno/.urlwatch/urls.yaml is group/world-writable

  Please remove shell jobs or fix these problems.

enno@dharma ~/git/urlwatch[extendable_urls_file*]$
```

Because of this I added this more complex code, that doesn't have that issue.
But of course in most cases this only happens once and the code would stay longer. It's a decision you as the maintainer have to make. I only want you to know all the facts.
I will apply your patch. I could revert it, if you change your mind.